examples/subscribing_to_stream.rs

```diff
@@ -88,16 +83,12 @@ pub async fn subscribe_to_all(client: &Client) -> Result<()> {
     // endregion subscribe-to-all-live
 
     // region subscribe-to-all-subscription-dropped
-    let mut stream = client.subscribe_to_all(&Default::default()).await?;
+    let options = SubscribeToAllOptions::default().auto_resubscribe_when_dropped();
```

Comment:
The naming looks good. It's explicit, and if we find a better name, we can easily provide the change in a non-breaking way (e.g. mark this one as obsolete, and create a new one).
We may consider renaming it to `resubscribe_when_dropped` and adding the retry setting as a param.